raster_loader/io.py

```diff
@@ -561,85 +401,52 @@ def rasterio_windows_to_records(
 
-            if output_quadbin:
-                rec = array_to_quadbin_record(
-                    raster_dataset.read(band, window=window),
-                    raster_dataset.nodata,
-                    band,
-                    band_name,
-                    str(band_type),
-                    transformer,
-                    raster_dataset.transform,
-                    resolution,
-                    window.row_off,
-                    window.col_off,
-                    crs=input_crs,
-                )
+            rec = array_to_quadbin_record(
+                raster_dataset.read(band, window=window),
+                raster_dataset.nodata,
+                band,
+                band_name,
+                str(band_type),
+                transformer,
+                raster_dataset.transform,
+                resolution,
+                window.row_off,
+                window.col_off,
+                crs=input_crs,
+            )
 
-            else:
-                rec = array_to_record(
-                    raster_dataset.read(band, window=window),
-                    raster_dataset.nodata,
-                    band,
-                    band_name,
-                    str(band_type),
-                    transformer,
-                    raster_dataset.transform,
-                    window.row_off,
-                    window.col_off,
-                    crs=input_crs,
-                    orientation=orientation,
-                    pseudo_planar=pseudo_planar,
-                )
-
-            metadata["crs"] = input_crs
-            metadata["gdal_transform"] = raster_dataset.transform.to_gdal()
-            metadata["nb_pixel_blocks"] += 1
-            metadata["nb_pixel"] += window.width * window.height
-            if (
-                not metadata["min_pixel_block_height_in_pixel"]
-                or metadata["min_pixel_block_height_in_pixel"] > window.height
-            ):
-                metadata["min_pixel_block_height_in_pixel"] = window.height
-            if metadata["max_pixel_block_height_in_pixel"] < window.height:
-                metadata["max_pixel_block_height_in_pixel"] = window.height
-            if (
-                not metadata["min_pixel_block_width_in_pixel"]
-                or metadata["min_pixel_block_width_in_pixel"] > window.width
-            ):
-                metadata["min_pixel_block_width_in_pixel"] = window.width
-            if metadata["max_pixel_block_width_in_pixel"] < window.width:
-                metadata["max_pixel_block_width_in_pixel"] = window.width
-            if (
-                window.height != metadata["max_pixel_block_height_in_pixel"]
-                or window.width != metadata["max_pixel_block_width_in_pixel"]
-            ):
-                metadata["irregular_pixel_block_shape"] = True
+            # metadata["nb_pixel_blocks"] += 1
+            # metadata["nb_pixel"] += window.width * window.height
+            metadata["num_blocks"] += 1
```

Comment:
This is not efficient, let's use the metadata to compute the num_blocks, so the num_pixels before hand

Reply:
I just followed the same logic as before, not clear why was in this way if there are some corner case to manage. BTW I'll remove.

Reply:
I guess there must be a way to get the num blocks from the raster info

Reply:
I've changed the formula using width and height